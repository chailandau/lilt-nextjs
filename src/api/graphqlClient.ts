import { GraphQLClient } from 'graphql-request';

const endpoint = process.env.NEXT_PUBLIC_PAYLOAD_SCHEMA as string;

export const graphQLClient = new GraphQLClient(endpoint, {
    headers: {
        Authorization: `users API-Key ${process.env.PAYLOAD_API_KEY}`
    }
});
