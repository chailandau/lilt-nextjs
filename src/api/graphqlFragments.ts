export const INTERNAL_LINK_FRAGMENT = `
internalLink {
    id
    title
    slug
}
`;

export const IMAGE_FRAGMENT = `
id
alt
url
width
height
base64
`;

export const VIDEO_FRAGMENT = `
id
url
placeholderImage {
    ${IMAGE_FRAGMENT}
}
`;

export const MENU_ITEMS_FRAGMENT = `
id
label
linkType
${INTERNAL_LINK_FRAGMENT}
externalLink
`;

export const CTA_FRAGMENT = `
id
label
linkType
${INTERNAL_LINK_FRAGMENT}
externalLink
color
`;

export const HERO_FRAGMENT = `
...on HeroBlock {
    __typename
    id
    hero {
        id
        heading
        subheading
        content
        cta {
            type
            ctaButtons {
                id
                callToAction {
                    ${CTA_FRAGMENT}
                }
            }
            ctaTiles {
                id
                callToActionTile {
                    id
                    linkType
                    label
                    ${INTERNAL_LINK_FRAGMENT}
                    externalLink
                    icon {
                        ${IMAGE_FRAGMENT}
                    }
                }
            }
        }
    }
}
`;
export const ACCORDION_FRAGMENT = `
...on AccordionBlock {
    __typename
    id
    accordion {
        id 
        heading
        accordionItems {
            id
            heading
            content
        }
    }
}
`;

export const CONVERSION_PANEL_FRAGMENT = `
...on ConversionPanelBlock {
    id
    __typename
    conversionPanel {
        id
        heading
        content
        callToAction {
            ${CTA_FRAGMENT}
        }
    }
}
`;

export const FEATURE_GRID_FRAGMENT = `
...on FeatureGridBlock {
    id
    __typename
    featureGrid {
        id 
        heading
        featureTiles {
            id
            heading
            image {
                ${IMAGE_FRAGMENT}
            }
            content
        }
    }
}
`;
export const ICON_TILE__GRID_FRAGMENT = `
...on IconTileGridBlock {
    id
    __typename
    iconTileGrid {
        id
        heading
        content
        iconTiles {
            id
            label
            icon {
                ${IMAGE_FRAGMENT}
            }
        }
        callToAction {
            ${CTA_FRAGMENT}
        }
    }
}
`;

export const PROCESS_FRAGMENT = `
...on ProcessBlock {
    id
    process {
        id
        steps {
            id
            heading
            content
        }
        callToAction {
            ${CTA_FRAGMENT}
        }
    }
}
`;

export const SWITCHBACK_FRAGMENT = `
...on SwitchbackBlock {
    __typename
    id
    switchback {
        id
        image {
            ${IMAGE_FRAGMENT}
        }
        imageSide
        heading
        content
    }
}
`;

const TEXT_GRID = `
 textGrid {
    id
    heading
    content
    textTiles {
        id
        text
    }
    callToAction {
        ${CTA_FRAGMENT}
    }
}
`;

export const TEXT_GRID_FRAGMENT = `
...on TextGridBlock {
    __typename
    id
    ${TEXT_GRID}
}
`;

export const TILE_GRID_FRAGMENT = `
...on TileGridBlock {
    id
    tileGrid {
        id
        heading 
        content
        tiles {
            id
            heading
            content
        }
        callToAction {
            ${CTA_FRAGMENT}
        }
    }
}
`;
export const FEATURED_MEDIA_FRAGMENT = `
...on FeaturedMediaBlock {
    id
    __typename
    featuredMedia {
        id
        heading 
        content
        assetType
        image {
            ${IMAGE_FRAGMENT}
        }
        video {
            ${VIDEO_FRAGMENT}
        }
        callToAction {
            ${CTA_FRAGMENT}
        }
    }
}
`;

export const SPECIALS_FRAGMENT = `
...on SpecialBlock {
    id
    __typename
    special {
      id
      heading
      subheading
      content
      sideBox
      ${TEXT_GRID}
    }
  }
`;
