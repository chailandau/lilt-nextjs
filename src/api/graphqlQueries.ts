import { gql } from 'graphql-request';

import {
    ACCORDION_FRAGMENT,
    CONVERSION_PANEL_FRAGMENT,
    CTA_FRAGMENT,
    FEATURED_MEDIA_FRAGMENT,
    FEATURE_GRID_FRAGMENT,
    HEADING_FRAGMENT,
    HERO_FRAGMENT,
    ICON_TILE__GRID_FRAGMENT,
    IMAGE_FRAGMENT,
    MENU_ITEMS_FRAGMENT,
    META_FRAGMENT,
    PROCESS_FRAGMENT,
    RICH_TEXT_FRAGMENT,
    SLUG_FRAGMENT,
    SPECIALS_FRAGMENT,
    SWITCHBACK_FRAGMENT,
    TEXT_GRID_FRAGMENT,
    TILE_GRID_FRAGMENT
} from './graphqlFragments';

export const PAGES_QUERY = gql`
query PagesQuery {
    Pages {
        ${SLUG_FRAGMENT}
    }
}
`;

export const LEGAL_PAGES_QUERY = gql`
query PagesQuery {
    LegalPages {
        ${SLUG_FRAGMENT}
    }
}
`;

export const PAGE_CONTENT_QUERY = gql`
query PageContentQuery($slug: String!) {
    Pages(where:{ slug: { equals: $slug }}) {
        docs {
            id
            slug
            title
            ${META_FRAGMENT}
            pageSections {
                enableGrid
                gridColor
                enableShapes
                shapes {
                    id
                    shape
                    location
                }
                blocks {
                    ${HERO_FRAGMENT}
                    ${ACCORDION_FRAGMENT}
                    ${CONVERSION_PANEL_FRAGMENT}
                    ${FEATURE_GRID_FRAGMENT}
                    ${FEATURED_MEDIA_FRAGMENT}
                    ${ICON_TILE__GRID_FRAGMENT}
                    ${PROCESS_FRAGMENT}
                    ${SPECIALS_FRAGMENT}
                    ${SWITCHBACK_FRAGMENT}
                    ${TEXT_GRID_FRAGMENT}
                    ${TILE_GRID_FRAGMENT}
                }
            }
            
        }
    }
}
`;

export const LEGAL_PAGE_CONTENT_QUERY = gql`
query LegalPageContentQuery($slug: String!) {
    LegalPages(where:{ slug: { equals: $slug }}) {
        docs {
            id
            title
            slug
            updatedAt
            legalBlocks {
                ${HEADING_FRAGMENT}
                ${RICH_TEXT_FRAGMENT}
            }
        }
    }
}
`;

export const HEADER_QUERY = gql`
query HeaderQuery  {
    Header {
        menuItems {
            ${MENU_ITEMS_FRAGMENT}
            submenuItems {
                ${MENU_ITEMS_FRAGMENT}
            }
        }
        callToAction {
            ${CTA_FRAGMENT}
        }
    }
}
`;

export const FOOTER_QUERY = gql`
query FooterQuery {
    Footer {
        socialLinks {
            icon {
                ${IMAGE_FRAGMENT}
            }
            url
            id
        }
        menuItems {
            ${MENU_ITEMS_FRAGMENT}
        }
        copyrightText
    }
}
`;
