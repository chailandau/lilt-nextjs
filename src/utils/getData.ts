import { graphQLClient } from '@/api/graphqlClient';
import { Query } from '@/api/graphqlTypes';

/**
 * Retrieves data from the API using the provided GraphQL query.
 *
 * @param QUERY - The GraphQL query to be executed.
 * @param slug - (Optional) The slug to be used as a variable in the query.
 * @returns A promise that resolves to the data retrieved from the API.
 */
export const getData = async (QUERY: string, slug: string | null = null) => {
    try {
        const variables = slug !== null ? { slug } : {};

        return await graphQLClient.request<Query>(QUERY, variables);
    } catch (error) {
        console.error('GraphQL request error:', error);
        throw error;
    }
};
