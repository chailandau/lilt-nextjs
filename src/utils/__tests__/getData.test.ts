import { getData } from '../getData';

import { graphQLClient } from '@/api/graphqlClient';
import { HEADER_QUERY } from '@/api/graphqlQueries';

jest.mock('@/api/graphqlClient', () => ({
    graphQLClient: {
        request: jest.fn()
    }
}));

describe('getData', () => {
    it('calls graphQLClient.request with the correct arguments', async () => {
        await getData(HEADER_QUERY);
        expect(graphQLClient.request).toHaveBeenCalledWith(HEADER_QUERY);
        expect(graphQLClient.request).toMatchSnapshot();
    });
});
