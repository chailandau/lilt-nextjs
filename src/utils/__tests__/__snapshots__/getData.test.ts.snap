// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getData calls graphQLClient.request with the correct arguments 1`] = `
[MockFunction] {
  "calls": [
    [
      "
    query HeaderQuery  {
            Header {
                menuItems {
                    
    id
    label
    linkType
    internalLink {
        
    id
    title
    slug

    }
    externalLink

                    submenuItems {
                        
    id
    label
    linkType
    internalLink {
        
    id
    title
    slug

    }
    externalLink

                    }
                }
                callToAction {
                    
    id
    label
    linkType
    internalLink {
        
    id
    title
    slug

    }
    externalLink
    color

            }
        }
    }
",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
